xbmc/addons/AddonManager.cpp

```diff
@@ -1024,4 +1054,44 @@ void CAddonMgr::FindAddons(ADDON_INFO_LIST& addonmap, const std::string& path)
   }
 }
 
+bool CAddonMgr::IsFromTrustedRepo(const AddonPtr& addon) const
+{
+  for (const auto& trustedRepo : m_trustedAddonRepos)
+  {
+    if (addon->Origin() == trustedRepo)
+      return true;
+  }
+
+  // not from a trusted repo
```

Comment:
most of the comments you have before `return true` and `return false` (here and in FindAddonAndCheckForUpdate) are quite obvious since your doxygen documentation for the method already makes it clean what the return value is.

Reply:
removed what's way too much